src/qbe_ir_generator.cpp

```diff
@@ -151,6 +151,35 @@ void QbeIrGenerator::Visit(const IfStmtNode& if_stmt) {
   output << end_label << std::endl;
 }
 
+void QbeIrGenerator::Visit(const WhileStmtNode& while_stmt) {
+  int label_num = NextLabelNum();
+  std::string body_label = PrefixLabel("loop_body", label_num);
+  std::string pred_label = PrefixLabel("pred", label_num);
+  std::string end_label = PrefixLabel("end", label_num);
+  int predicate_num = 0;
+
+  // While statement evaluates predicate before body statement.
+  // Do While statement evaluates predicate after body statement.
```

Comment:
Suggest the following comment to tell this non-trivial story:
```suggestion
  // In a while statement, the predicate is evaluated before the body statement,
  // whereas in a do-while statement, the order is reversed.
  // In the generated code for a while statement, the predicate precedes the
  // body, and, in addition to the conditional jump after the predicate, there
  // is an additional unconditional jump at the end of the body to loop back to
  // the predicate. On the other hand, a do-while statement has its predicate
  // after the body, thus requiring only a conditional jump.
```